csharp/src/Microsoft.ML.OnnxRuntime/NativeMethods.cs

```diff
@@ -192,6 +192,22 @@ public struct OrtApi
         public IntPtr ModelMetadataGetGraphDescription;
     }
 
+    #region ORT Provider options
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct OrtCUDAProviderOptionsNative
+    {
+        public int device_id;                                  // cuda device with id=0 as default device.
```

Comment:
>int [](start = 15, length = 3)

I would want to verify that this is being marshalled correctly. In C# int is always 4 bytes. In the native code on Windows it is 4 bytes but on Linux it is 8 bytes. So it might work OK on windows but we need it work everywhere.

Reply:
No. Even on Linux int is still 4 bytes. You were talking 'long'

Reply:
You are correct.

---
In reply to: [613471164](https://github.com/microsoft/onnxruntime/pull/7315#discussion_r613471164) [](ancestors = 613471164)

Reply:
We may have an issue here. The new C API was added without giving a though how this is going to interact with other languages even though the purpose of having C API is language binding.

---
In reply to: [613468046](https://github.com/microsoft/onnxruntime/pull/7315#discussion_r613468046) [](ancestors = 613468046)